### PR TITLE
Renamed titles, added links and improved wording

### PR DESCRIPTION
**Pull request by aleon1220**
changed the title for the 3 options on how to set up AWS IAM structure
1-with CF console
2-With AWS CLI
3-with the AWS IAM console
Listed the main components of the AWS IAM structure
In halyard section changed titles to put them in context by using the AWS provider.
I changed the wording to start with actions to perform e.g. Enable the AWS Provider

In Next steps section
I added links to the other AWS compute services: ECS, EKS and lambda

### CODE REVIEW

File: setup/install/providers/aws/aws-ec2.md [hunk 1]
```diff
@@ -11,13 +11,19 @@ sidebar:
 > worked on. In the meantime, please use the following
 > [AWS tutorial: Continuous Delivery using Spinnaker on Amazon EKS](https://aws.amazon.com/blogs/opensource/continuous-delivery-spinnaker-amazon-eks/).
 
-Use the AWS EC2 Provider if you want to manage EC2 Instances via Spinnaker. Refer to the [AWS Cloud Provider Overview](https://spinnaker.io/setup/install/providers/aws/) to understand how AWS IAM must be set up for the AWS EC2 provider to work.
+Use the AWS EC2 Provider if you want to manage EC2 Instances via Spinnaker. Refer to the [AWS Cloud Provider Overview](https://spinnaker.io/setup/install/providers/aws/) to understand how AWS IAM must be set up with the Spinnaker AWS EC2 provider.
+
+Spinnaker will use an [AWS IAM structure](https://aws.amazon.com/iam/) of users, roles, policies, etc to access AWS services and resources securely. There are 3 options to set up the AWS IAM structure
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]:
```suggestion
Spinnaker will use an [AWS IAM structure](https://aws.amazon.com/iam/) of users, roles, policies, and so on, to access AWS services and resources securely. There are 3 options to set up the AWS IAM structure
```

File: setup/install/providers/aws/aws-ec2.md [hunk 2]
```diff
@@ -11,13 +11,19 @@ sidebar:
...
+
+1. AWS CloudFormation templates deployed with the CloudFormation Console
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:18]: Maybe hyperlink these to their respective headings below?

File: setup/install/providers/aws/aws-ec2.md [hunk 3]
```diff
@@ -11,13 +11,19 @@ sidebar:
...
+2. AWS CloudFormation templates deployed with AWS CLI
+3. Manually creating the IAM structure with the AWS IAM Console
 
 In [AWS](https://aws.amazon.com/){:target="\_blank"}, an [__Account__](/concepts/providers/#accounts)
 maps to a credential able to authenticate against a given [AWS
 account](https://aws.amazon.com/account/){:target="\_blank"}.
 
-## Option-1 : Use AWS Console to configure AWS
+## Option-1 : Configure with AWS CloudFormation Console
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]:
```suggestion
## Option 1: Configure with AWS CloudFormation Console
```

File: setup/install/providers/aws/aws-ec2.md [hunk 4]
```diff
@@ -45,7 +51,7 @@ Use this option to deploy Spinnaker, if you are familar with deployment using [A
 
 ![](../outputs_cloudformation.png)
 
-## Option-2 : Use AWS CLI to configure AWS
+## Option-2 : Configure with AWS CLI
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]:
```suggestion
## Option 2: Configure with AWS CLI
```

File: setup/install/providers/aws/aws-ec2.md [hunk 5]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
 --parameter-overrides AuthArn=FROM_ABOVE ManagingAccountId=FROM_ABOVE --capabilities CAPABILITY_NAMED_IAM --region us-west-2
 ```
 
-## Option-3 : Use AWS Console UI (Manual Steps) 
+## Option-3 : Configure with AWS IAM Console
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:94]:
```suggestion
## Option 3: Configure with AWS IAM Console
```

File: setup/install/providers/aws/aws-ec2.md [hunk 6]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
...
+### Create Roles
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:95]: Are these h3-level sections going to be filled in later?

File: setup/install/providers/aws/aws-ec2.md [hunk 7]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
...
+
+### Create Policies
+
+
+### Associate Policies to User or Roles
+The permissions defined in the policies are attached to the users or roles. This will allows Spinnaker to use either a user or a role and deploy EC2 instances in any particular region.
 
 There are 2 options here
-1. Using AWS IAM AccessKey and Secret
+1. Using AWS IAM User with AccessKey and Secret
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:105]:
```suggestion
1. Use AWS IAM User with AccessKey and Secret
```

File: setup/install/providers/aws/aws-ec2.md [hunk 8]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
...
 Option number 1 is useful for creation of user with AWS Access Key and secret. This is a common configuration. 
 2. Using AWS IAM Roles
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:107]:
```suggestion
2. Use AWS IAM Roles
```

File: setup/install/providers/aws/aws-ec2.md [hunk 9]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
...
 Option 2 uses the IAM roles *ManagingRole* and *ManagedRoles*. This setting is applied on some environments that have extra security considerations.
 
 ## Halyard Configurations
-After the AWS IAM user, roles, policies and trust relationship have been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
+After the AWS IAM  structure (user, roles, policies and trust relationship) has been set up, the next step is to add the AWS configurations to Spinnaker via Halyard CLI:
 
+The General steps are the following:
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:113]: You can omit this sentence.

File: setup/install/providers/aws/aws-ec2.md [hunk 10]
```diff
@@ -85,22 +91,31 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
...
 1. Access the Halyard Pod.
 2. Add the configurations for AWS provider with `hal` command. Please check [hal config provider AWS](https://www.spinnaker.io/reference/halyard/commands/#hal-config-provider-aws).
 3. Enable the AWS provider `hal config provider aws enable`.
 
-### Configure Halyard to use AccessKeys (if configured)
+### Configure Spinnaker AWS provider to use AccessKeys (if using AWS IAM user)
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:118]:
```suggestion
### Configure the Spinnaker AWS provider to use AccessKeys (if using AWS IAM user)
```

Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:118]: I like these changes a lot, @aleon1220 ! You don't "configure Halyard," you use Halyard to configure Spinnaker!

File: setup/install/providers/aws/aws-ec2.md [hunk 11]
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
...
 
-Now enable AWS
+### Enable the Spinnaker AWS provider
 
 ```bash
 hal config provider aws enable
 ```
 
 ## Advanced account settings
+Once you configure the Spinnaker AWS provider you will be able to manage and deploy EC2 resources with Spinnaker.
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]:
```suggestion
After you configure the Spinnaker AWS provider you can manage and deploy EC2 resources with Spinnaker.
```

File: setup/install/providers/aws/aws-ec2.md [hunk 12]
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
...
 
-You can view the available configuration flags for AWS within the
-[Halyard reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
+You can view the available configuration flags for Spinnaker AWS provider within the
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md:146]:
```suggestion
You can view the available configuration flags for the Spinnaker AWS provider in the
```

File: setup/install/providers/aws/aws-ec2.md [hunk 13]
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
...
+[Halyard AWS provider reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]:
```suggestion
the [Halyard command reference](/reference/halyard/commands#hal-config-provider-aws-account-add).
```

File: setup/install/providers/aws/aws-ec2.md [hunk 14]
```diff
@@ -119,21 +134,24 @@ hal config provider aws account add $AWS_ACCOUNT_NAME \
...
 
 ## Next steps
 
-Optionally, you can [set up Amazon's Elastic Container
-Service](/setup/install/providers/aws/aws-ecs/) or [set up another cloud
-provider](/setup/install/providers/), but otherwise you're ready to
-[choose an environment](/setup/install/environment/)
+Optionally, you can enable other AWS Providers:
+[set up Amazon's Elastic Container
```
Comment (dorbin) [inline setup/install/providers/aws/aws-ec2.md]: Andres, can you set these as a bulleted list?

**Review by dorbin — COMMENTED** [review]
Nice! I left a bunch of edits though. :)

File: setup/install/providers/aws/aws-ec2.md (hunk 11 above)
Comment (aleon1220) [reply setup/install/providers/aws/aws-ec2.md]: Thanks. More proper English there. Thank you.

File: setup/install/providers/aws/aws-ec2.md (hunk 14 above)
Comment (aleon1220) [reply setup/install/providers/aws/aws-ec2.md]: yes

**Review by aleon1220 — COMMENTED** [review]
updated changes.

**Comment by aleon1220** [comment]
@dorbin Thanks Dave for your suggestions. I have planned some extra improvements to the complete AWS Provider (EC2, ECS, EKS, Lambda). I have been doing progressively and in small chunks. Thanks.
If you have improvement ideas i am happy to read about them.

File: setup/install/providers/aws/aws-ec2.md (hunk 2 above)
Comment (dorbin) [reply setup/install/providers/aws/aws-ec2.md:18]: I meant link each of these three list items to the headings below, in the article (though with headings like these, the auto-generated anchor will be complex. The current anchor for option 1 (on the published site, before these changes) is `#option-1--use-aws-console-to-configure-aws`

**Review by dorbin — APPROVED** [review]

**Comment by aleon1220** [comment]
OK I have created a new PR https://github.com/spinnaker/spinnaker.github.io/pull/1996 with the changes from this PR1964
Hopefully it can be easily merged...

**Pull request closed without merging**